WeakAuras/Prototypes.lua

```diff
@@ -6351,17 +6351,74 @@ Private.event_prototypes = {
     internal_events = { "WA_DELAYED_PLAYER_ENTERING_WORLD", },
     force_events = "UNIT_INVENTORY_CHANGED",
     name = L["Item Bonus Id Equipped"],
+    statesParameter = "one",
+    init = function(trigger)
+      local ret = [=[
+        local itemBonusId, itemId, itemName, itemIcon, itemSlot, itemSlotString = WeakAuras.GetBonusIdInfo(%q)
+        local itemBonusId = tonumber(itemBonusId)
+        local icon = %s and WeakAuras.GetLegendaryIcon(itemBonusId) or itemIcon
+      ]=]
+      return ret:format(trigger.itemBonusId, trigger.use_legendaryIcon and "true" or "false")
+    end,
     args = {
       {
         name = "itemBonusId",
         display = L["Item Bonus Id"],
         type = "string",
-        test = "WeakAuras.CheckForItemBonusId(%q)",
+        store = "true",
+        test = "itemBonusId",
         required = true,
         desc = function()
           return WeakAuras.GetLegendariesBonusIds()
           .. L["\n\nSupports multiple entries, separated by commas"]
-        end
+        end,
+        conditionType = "number",
+      },
+      {
+        name = "legendaryIcon",
+        display = L["Use Legendary Effect Icon instead of Item Icon if possible"],
+        type = "toggle",
+        test = "true",
+        desc = L["Legendary effects have their own icon. Toggle this to display that icon instead of the icon of the item providing you the bonus id."],
+      },
+      {
+        name = "name",
+        display = L["Item Name"],
+        hidden = "true",
+        init = "itemName",
+        store = "true",
+        test = "true",
+        conditionType = "string",
+      },
+      {
+        name = "icon",
+        hidden = "true",
+        init = "icon or 'Interface\\Icons\\INV_Misc_QuestionMark'",
+        store = "true",
+        test = "true",
+      },
+      {
+        name = "itemId",
+        display = L["Item Id"],
+        hidden = "true",
+        store = "true",
+        test = "true",
+        conditionType = "number",
```

Comment:
For id I was considering a use case where the bonus id could match more than one item (say you craft a legendary helm and belt with the same bonus and want to distinguish the two without using multiple triggers). It could potentially be covered by the slot condition but I wanted to provide options, which kind of led to name being included as well..

Reply:
I don't actually _know_ what other effects in the game give a bonus id, which was what led to me casting a wider net on available options.

Theoretically you could have the same bonus id on the same slot on two different pieces, as far as I'm aware. I don't know if that's actually possible in-game but it was something I considered.

Name has less compelling arguments for it though.

Reply:
Bonus ids are all kind of things, and it might be useful in the future. 

On the item id, if you add 

operator_types = "only_equal

then, at least it'll only allow checking for equality, which is a bit saner at least.

Reply:
Technically, I didn't actually limit it to bonus ids, so anything on: https://wow.gamepedia.com/ItemString works.

Now, 99% of that doesn't seem usefull to me though.